### PR TITLE
Replace slacker with slack_sdk

### PR DESCRIPTION
**Pull request by rebkwok**
This is the quick fix to address https://github.com/ebmdatalab/ebmbot/issues/208

Slackbot appears to no longer be maintained; this incorporates the changes in  [this PR](https://github.com/scrapinghub/slackbot/pull/229), with a few additional updates to API method calls that were missed there.

The functional tests no longer run (unless you have an old bot integration to test with) because of the availability of the legacy test user tokens.  Github actions are set up to run the unit tests only.

In the longer term, we'll probably want to migrate ebmbot to use the events API, but this should ensure it continues working after [next month](https://api.slack.com/changelog/2021-10-rtm-start-to-stop)

### CODE REVIEW

File: slackbot/slackclient.py [hunk 1]
```diff
@@ -61,22 +66,35 @@ def parse_slack_login_data(self, login_data):
         self.login_data = login_data
         self.domain = self.login_data['team']['domain']
         self.username = self.login_data['self']['name']
-        self.parse_user_data(login_data['users'])
-        self.parse_channel_data(login_data['channels'])
-        self.parse_channel_data(login_data['groups'])
-        self.parse_channel_data(login_data['ims'])
```
Comment (StevenMaude) [inline slackbot/slackclient.py:67]: question: Below, `parse_channel_data()` is no longer getting called for the equivalent of `groups` and `ims`, but we still do make these calls for users (`parse_user_data()`) and channels? Is this OK? (It probably is.)

**Comment by StevenMaude** [comment]
suggestion: There's a few things that could *optionally* be tidied up maybe if we're diverging this from the upstream :sparkles:

* The `tox.ini` refers to `py36` and should be updated or dropped.
* The `requirements.txt` isn't made with `pip-compile`. It would be nice if we specified this is in the same way as we do with most of our other repositories.
* The `pytest.ini` could potentially updated to ignore the functional tests, if we can never run them again. That way, you shouldn't have to specify as such in the GitHub workflow, and there's only one place the tests are configured.
* `travis.yml` can be deleted.
* The code is using `six` in a few places, but doesn't even support Python 2 any more, so could be removed. I don't think this is necessarily worth bothering with though.

Feel free to pick and choose from this list, or ignore entirely.

**Comment by StevenMaude** [comment]
Otherwise, looks fine and great that you've resolved this ahead of the required date for the API changes :+1:

File: slackbot/slackclient.py (hunk 1 above)
Comment (rebkwok) [reply slackbot/slackclient.py:67]: Yup, this is channels, groups and ims are all now `conversations` in slack_sdk; later in this method we do: 
```
    for page in self.webapi.conversations_list(
                exclude_archived=True,
                types="public_channel,private_channel",
                limit=1000
        ):
            self.parse_channel_data(page['channels'])
```
That covers channels and groups (which is what slacker called private conversations).  It doesn't include IMs, and I think it should - I'll update it to inlcude ` types="public_channel,private_channel,im,mpim"`, which will include direct messages and group dms

**Comment by rebkwok** [comment]
> suggestion: There's a few things that could _optionally_ be tidied up maybe if we're diverging this from the upstream sparkles
> 
> * The `tox.ini` refers to `py36` and should be updated or dropped.
> * The `requirements.txt` isn't made with `pip-compile`. It would be nice if we specified this is in the same way as we do with most of our other repositories: that is, pinned the versions used in deployment, with hashes.
> * The `pytest.ini` could potentially updated to ignore the functional tests, if we can never run them again. That way, you shouldn't have to specify as such in the GitHub workflow, and there's only one place the tests are configured.
> * `travis.yml` can be deleted.
> * The code is using `six` in a few places, but doesn't even support Python 2 any more, so could be removed. I don't think this is necessarily worth bothering with though.
> 
> Feel free to pick and choose from this list, or ignore entirely.

So I thought about most of these, and then decided to do the minimal thing, partly because I was planning on replacing this whole thing anyway.  Given that it might not be that soon, I'll take another look.

File: slackbot/slackclient.py (hunk 1 above)
Comment (StevenMaude) [reply slackbot/slackclient.py:67]: That makes sense; thanks for the clarification!

**Comment by StevenMaude** [comment]
> So I thought about most of these, and then decided to do the minimal thing, partly because I was planning on replacing this whole thing anyway. Given that it might not be that soon, I'll take another look.

I don't think it particularly blocks this PR: it's definitely better to merge this sooner, instead of letting it languish and catch us out when the current deployment breaks. Just that I spotted them while looking around.

Unfortunately, because this is a fork, you don't get your own issue tracker to note tasks for later. (I think you can if you request to "unfork" via GitHub Support).

**Comment by rebkwok** [comment]
> Feel free to pick and choose from this list, or ignore entirely.
I've done all of them I think.  Makes sense to ditch python 2 and six.  I've unpinned the requirements in the setup.py, which I think is OK (at least, we don't seem to pin setup.py requirements in other repos, and my test bot is working fine with a clean venv install).

**Comment by StevenMaude** [comment]
> I've done all of them I think.

Great! All looks good :heavy_check_mark:

**Pull request merged**